scripts/build/Jenkins/Jenkinsfile

```diff
@@ -586,24 +609,26 @@ def CreateSingleNode(Map pipelineConfig, def platform, def build_job, Map envVar
                         if (build_job_name.toLowerCase().contains('asset') && env.IS_UPLOAD_AP_LOGS?.toBoolean()) {
                             CreateUploadAPLogsStage(platform.key, build_job_name, envVars['WORKSPACE'], platform.value.build_types[build_job_name].PARAMETERS).call()
                         }
+                        // Upload test artifacts only on builds that failed and ran test suites
+                        if (params.containsKey('CMAKE_TARGET') && params.CMAKE_TARGET.contains("TEST_SUITE")) {
```

Comment:
You need to add an EVN check here just like "env.IS_UPLOAD_AP_LOGS", so we can enable/disable this feature for different Jenkins environment, I would suggest run this in LY Fork Jenkins for a week before enabling this in O3DE Jenkins

Reply:
added env.IS_UPLOAD_TEST_ARTIFACTS check. Will this need to be added on the Jenkins side? I will run this in a fork jenkins first as suggested.